src/System.Management.Automation/FormatAndOutput/common/PSStyle.cs

```diff
@@ -348,29 +412,87 @@ public sealed class FileInfoFormatting
             /// <summary>
             /// Gets or sets the style for directories.
             /// </summary>
-            public string Directory { get; set; } = "\x1b[44;1m";
+            public string Directory
+            {
+                get => _directory;
+                set => _directory = ValidateNoContent(value);
+            }
+
+            private string _directory = "\x1b[44;1m";
 
             /// <summary>
             /// Gets or sets the style for symbolic links.
             /// </summary>
-            public string SymbolicLink { get; set; } = "\x1b[36;1m";
+            public string SymbolicLink
+            {
+                get => _symbolicLink;
+                set => _symbolicLink = ValidateNoContent(value);
+            }
+
+            private string _symbolicLink = "\x1b[36;1m";
 
             /// <summary>
             /// Gets or sets the style for executables.
             /// </summary>
-            public string Executable { get; set; } = "\x1b[32;1m";
+            public string Executable
+            {
+                get => _executable;
+                set => _executable = ValidateNoContent(value);
+            }
+
+            private string _executable = "\x1b[32;1m";
+
+            /// <summary>
+            /// Custom dictionary handling validation of extension and content.
+            /// </summary>
+            public sealed class FileExtensionDictionary : Dictionary<string, string>
+            {
+                /// <summary>
+                /// Initializes a new instance of the <see cref="FileExtensionDictionary"/> class.
+                /// </summary>
+                public FileExtensionDictionary() : base(StringComparer.OrdinalIgnoreCase) { }
+
+                /// <summary>
+                /// Add new extension and decoration to dictionary.
+                /// </summary>
+                /// <param name="extension">Extension to add.</param>
+                /// <param name="decoration">ANSI string value to add.</param>
+                public new void Add(string extension, string decoration)
```

Comment:
This is dangerous, since handling this type as a `Dictionary<string, string>` or possibly as other types won't use your logic.

See https://stackoverflow.com/a/6250932

Reply:
The class is now sealed.

Reply:
will change implementation